File: gugu.js
function encode(o) {
    return Array.from(o)
        .map(n => n.charCodeAt(0).toString(16).padStart(4, "0"))
        .map(n => Array.from(n)
            .map(e => "о" + String.fromCharCode(parseInt("030" + e, 16)) + "古")
            .join(""))
        .join("")
}

function decode(o) {
    return Array.from(o.split(/(?=(?:............)*$)/))
        .map(n => Array.from(n)
            .filter(e => e !== "о" && e !== "古")
            .map(e => (e.charCodeAt(0) - 0x0300).toString(16))
            .join(""))
        .map(n => String.fromCharCode(parseInt(n, 16)))
        .join("")
}

File: gugu.py
def encode(o: str) -> str:
    return "".join(
        "".join(
            "о" + chr(int("030" + e, 16)) + "古"
            for e in f"{ord(ch):04x}"
        )
        for ch in o.strip()
    )


def decode(o: str) -> str:
    parts = [o[i:i + 12] for i in range(0, len(o.strip()), 12)]
    return "".join(
        chr(int("".join(
            hex(ord(e) - 0x0300)[2:]
            for e in n if e not in ("о", "古")
        ), 16))
        for n in parts
    )
